### ISSUE TITLE
`redundant_async_block` + moved values

### ISSUE THREAD

**Issue by cynecx** [rust-lang/rust-clippy#10507]
### Summary

The `redundant_async_block` lint triggers even when the closure captures locals by moving them. I've noticed multiple instances of these false-positives in our code-base.

### Lint Name

redundant_async_block

### Reproducer

I tried this code:

```rust
use std::future::Future;

async fn work(_: &str) {}

fn spawn(_: impl Future<Output = ()> + 'static) {}

fn main() {
    let val = "Hello World".to_owned();
    spawn(async move { work(&val).await });
}
```

I saw this happen:

```
warning: this async expression only awaits a single future
 --> src/main.rs:9:11
  |
9 |     spawn(async move { work(&val).await });
  |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: you can reduce it to: `work(&val)`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_async_block
  = note: `#[warn(clippy::redundant_async_block)]` on by default

warning: `test-clippy-1` (bin "test-clippy-1") generated 1 warning (run `cargo clippy --fix --bin "test-clippy-1"` to apply 1 suggestion)
```

I expected to see this happen:

No lint.

### Version

```text
rustc 1.70.0-nightly (171693274 2023-03-14)
binary: rustc
commit-hash: 1716932743a7b3705cbf0c34db0c4e070ed1930d
commit-date: 2023-03-14
host: aarch64-apple-darwin
release: 1.70.0-nightly
LLVM version: 15.0.7
```

**Comment by mk12** [comment]
Looks like a duplicate of #10482.

**Comment by cynecx** [comment]
Sorry about that.
